DataBiosphere/leonardo
# [IA-4363] Don't filter by Google Project for Azure disks

__Jira ticket__: https://broadworkbench.atlassian.net/browse/[ticket_number]




## Summary of changes

Fix for [this thread
](https://broadinstitute.slack.com/archives/CA3NP1733/p1693243543449879)

Partial revert for [this PR](https://github.com/DataBiosphere/leonardo/pull/3552)

### What

- Partitioning by type (azure vs gcp)
- filter by project for GCP
- filter the old way for azure (with the `filterUserVisibleWithProjectFallback` function)
- add them back together

### Why

- Changes to the V1 listDisks endpoint to check user access to the parent google project filtered out Azure disks

## Testing these changes

### What to test

- Create an Azure VM
- Look for PD in cloud environment page
- Add a file to PD folder
- Delete the VM and keep the disk
- recreate the VM, make sure the disk settings are grayed out this message is displayed
<img width="663" alt="image" src="https://github.com/DataBiosphere/leonardo/assets/85642387/b41ed27a-ec44-4b0c-bdea-97548dbad769">

- Check that file is still there on recreation



### Who tested and where

  - _NB: Rerun automation tests on this PR by commenting `jenkins retest` or `jenkins multi-test`._
